app.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'sinatra/reloader'
+require 'erb'
+require 'json'
+require 'securerandom'
+
+def load_memos
+  json_data = File.open('memos.json') { |file| JSON.parse(file.read) }
+  json_data['memos']
+end
+
+def load_memo
+  memos = load_memos
+  @memo = memos.find { |memo| memo['id'] == params[:id] }
+end
+
+def save_memos(memos)
+  File.open('memos.json', 'w') { |file| JSON.dump({ 'memos' => memos }, file) }
+end
+
+helpers do
+  def h(text)
+    Rack::Utils.escape_html(text)
+  end
+end
+
+not_found do
+  status 404
+  erb :not_found
+end
+
+get '/memos' do
+  @memos = load_memos
+  erb :memos
+end
+
+get '/memos/new' do
+  erb :new_memo
+end
+
+post '/memos' do
+  title = params[:title]
+  text = params[:text]
+  id = SecureRandom.uuid
+  new_memo = { 'id' => id, 'title' => title, 'text' => text }
+  memos = load_memos
+  memos << new_memo
+  save_memos(memos)
+  redirect "/memos/#{new_memo['id']}"
```

Comment:
意味のまとまり的に、以下のように空行を入れたほうがいいかなーと思います。必要であれば、他のルーティングハンドラーでもそうしてください。コメントは、説明のために入れているだけなので、プログラム中に残す必要はありません。

```suggestion
  # 新規メモのデータの作成
  title = params[:title]
  text = params[:text]
  id = SecureRandom.uuid
  new_memo = { 'id' => id, 'title' => title, 'text' => text }
  
  # 新規メモのデータの保存
  memos = load_memos
  memos << new_memo
  save_memos(memos)
  
  # 完了後の画面遷移
  redirect "/memos/#{new_memo['id']}"
```

Reply:
ありがとうございます。
とても分かりやすいので参考にさせていただきます。
他のルーティングハンドラーも修正いたします。

まとまりを分ける際にどういうところに気を付ければ良いか教えていだたいても良いでしょうか？
`patch '/memos/:id' do`だと
```
patch '/memos/:id' do
  # メモのデータの読み込み
  memos = load_memos

  # 更新対象のメモを見つける
  target_memo = memos.find { |memo| memo['id'] == params[:id] }

  # メモの内容を更新
  target_memo['title'] = params[:title]
  target_memo['text'] = params[:text]

  # 更新したメモのデータを保存
  save_memos(memos)

  # 更新したメモの詳細ページにリダイレクト
  redirect "/memos/#{params[:id]}"
end
```
や
```
patch '/memos/:id' do
  # メモデータの読み込み
  memos = load_memos
  target_memo = memos.find { |memo| memo['id'] == params[:id] }

  # メモの内容を保存
  target_memo['title'] = params[:title]
  target_memo['text'] = params[:text]
  save_memos(memos)

  # 更新したメモの詳細ページにリダイレクト
  redirect "/memos/#{params[:id]}"
end
```
みたいにもできるのかと思ったのでお聞きしました。

Reply:
> まとまりを分ける際にどういうところに気を付ければ良いか教えていだたいても良いでしょうか？

明確な基準は特にないですねー。前者のような分け方でもよいと思います。人それぞれだったり、プロジェクトによったりすると思います。

Reply:
ありがとうございます。
上のだと細かく分けることもできそうだったので気になってお聞きしました。